app/src/main/java/com/mad41/tripreminder/room_database/view_model/TripRepository.java

```diff
@@ -0,0 +1,163 @@
+package com.mad41.tripreminder.room_database.view_model;
+
+import android.app.Application;
+import android.os.AsyncTask;
+
+import androidx.lifecycle.LiveData;
+
+import com.mad41.tripreminder.room_database.MyRoomDataBase;
+import com.mad41.tripreminder.room_database.trip.Trip;
+import com.mad41.tripreminder.room_database.trip.TripDao;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+public class TripRepository {
+    private TripDao tripDao;
+    private LiveData<List<Trip>> allTrips;
+    private LiveData<List<Trip>> upcomingTrips;
+    private LiveData<List<Trip>> historyTrips;
+
+    public TripRepository(Application application) {
+        MyRoomDataBase database = MyRoomDataBase.getUserDataBaseInstance(application);
+        tripDao = database.tripDao();
+        allTrips=tripDao.getAllTrips();
+        upcomingTrips=tripDao.getUpcomingTrips();
+        historyTrips=tripDao.getHistoryTrips();
+
+    }
+    public long insertTrip(Trip trip) {
+        long id=0;
+        try {
+             id= new InsertTripAsyncTask(tripDao).execute(trip).get();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return id;
+    }
+    public void updateStatus(int id,int status) {
+        new updateStatusAsyncTask(tripDao).execute(id,status);
+    }
+    public void updateTrip(Trip note) {
+        new UpdateTripAsyncTask(tripDao).execute(note);
+    }
+    public void deleteTrip(Trip note) {
+        new UpdateTripAsyncTask(tripDao).execute(note);
+    }
+    public void deleteTripById(int id) {
+        new deleteTripByIdAsyncTask(tripDao).execute(id);
+    }
+    public void deleteAllTrips() {
+        new DeleteAllTripsAsyncTask(tripDao).execute();
+    }
+    public Trip getTripById(int id) {
+        Trip trip=null;
+        try {
+             trip= new getTripByIdAsyncTask(tripDao).execute(id).get();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        return trip;
+    }
+
+    public LiveData<List<Trip>> getAllNotes() {
+        return allTrips;
+    }
+    public LiveData<List<Trip>> getUpcomingNotes() {
+        return upcomingTrips;
+    }
+    public LiveData<List<Trip>> getHistoryNotes() {
+        return historyTrips;
+    }
+
+    private static class updateStatusAsyncTask extends AsyncTask<Integer, Void, Void> {
```

Comment:
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Class names should start with an upper case letter (NM_CLASS_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-should-start-with-an-upper-case-letter-nm-class-naming-convention).
Class names should be nouns, in mixed case with the first letter of each internal word capitalized. Try to keep your class names simple and descriptive. Use whole words-avoid acronyms and abbreviations (unless the abbreviation is much more widely used than the long form, such as URL or HTML).